src/containers/SortByContainer.tsx

```diff
@@ -0,0 +1,41 @@
+import { connect} from "react-redux";
+import * as React from "react";
+import { Store, searchParameterActions, asyncActions } from "azsearchstore";
+import * as redux from "redux";
+import SortBy from "../components/SortBy";
+
+function getReturnType<RT>(expression: (...params: any[]) => RT): RT {
+    return {} as RT;
+}
+
+export interface OwnProps {
+  fields: [{title: string, field: string}];
+  defaultField: string;
+  css: { [key: string]: string };
+}
+
+const mapDispatchToProps = (dispatch: redux.Dispatch<any>, ownProps: OwnProps) => {
+  return {
+    onSortChange: (field: string, direction: string) => {
+      dispatch(searchParameterActions.updateSearchParameters({orderby: field + (field !== "" ? " " + direction : "")}));
```

Comment:
Maybe we can refactor this to make it a little more readable? if field is empty string, it will evaluate as false. [TypeScript has a neat feature called object destructuring](https://basarat.gitbooks.io/typescript/docs/destructuring.html) which helps when the name of the object key, and the name of the variable holding the value are the same.

```ts
const orderBy = field ? `${field} ${direction}` : "";
dispatch(searchParameterActions.updateSearchParameters({ orderby }));
```